From 73059ff9b63c364bfe52acf05f521e869ca524cf Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Fri, 21 Apr 2023 10:52:18 +0200
Subject: [PATCH 295/954] feat(stm32mp2-fdts): update SoC device tree

Move all secure-able peripherals under RIFSC bus node. This is an
alignment with OP-TEE device tree file.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 0f6d1fd1758cb031070b9cf570ff383e03cfb17a feat(stm32mp2-fdts): update SoC device tree]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: I34a893c855471a1002845067e72d4e48b4a4c306
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/302116
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 fdts/stm32mp25-bl2.dtsi  |   6 +-
 fdts/stm32mp25-bl31.dtsi |   6 +-
 fdts/stm32mp251.dtsi     | 150 +++++++++++++++++++--------------------
 3 files changed, 83 insertions(+), 79 deletions(-)

diff --git a/fdts/stm32mp25-bl2.dtsi b/fdts/stm32mp25-bl2.dtsi
index 3f27282431..eb91c1c889 100644
--- a/fdts/stm32mp25-bl2.dtsi
+++ b/fdts/stm32mp25-bl2.dtsi
@@ -18,8 +18,10 @@
 		/delete-node/ memory-controller@48200000;
 #endif
 #if !STM32MP_EMMC && !STM32MP_SDMMC
-		/delete-node/ mmc@48220000;
-		/delete-node/ mmc@48230000;
+		rifsc@42080000 {
+			/delete-node/ mmc@48220000;
+			/delete-node/ mmc@48230000;
+		};
 #endif
 	};
 
diff --git a/fdts/stm32mp25-bl31.dtsi b/fdts/stm32mp25-bl31.dtsi
index fc9e4de078..79703749bd 100644
--- a/fdts/stm32mp25-bl31.dtsi
+++ b/fdts/stm32mp25-bl31.dtsi
@@ -7,7 +7,9 @@
 	soc@0 {
 		/delete-node/ ommanager@40500000;
 		/delete-node/ memory-controller@48200000;
-		/delete-node/ mmc@48220000;
-		/delete-node/ mmc@48230000;
+		rifsc@42080000 {
+			/delete-node/ mmc@48220000;
+			/delete-node/ mmc@48230000;
+		};
 	};
 };
diff --git a/fdts/stm32mp251.dtsi b/fdts/stm32mp251.dtsi
index 2fc364c196..cb8da66223 100644
--- a/fdts/stm32mp251.dtsi
+++ b/fdts/stm32mp251.dtsi
@@ -84,8 +84,38 @@
 		interrupt-parent = <&intc>;
 		ranges = <0x0 0x0 0x0 0x80000000>;
 
+		ommanager: ommanager@40500000 {
+			#address-cells = <2>;
+			#size-cells = <1>;
+			compatible = "st,stm32mp25-omm";
+			reg = <0x40500000 0x400>, <0x60000000 0x10000000>;
+			reg-names = "omm", "omm_mm";
+			clocks = <&rcc CK_BUS_OSPIIOM>;
+			resets = <&rcc OSPIIOM_R>;
+			status = "disabled";
+
+			ranges = <0 0 0x40430000 0x400>,
+				 <1 0 0x40440000 0x400>;
+
+			ospi1: spi@40430000 {
+				compatible = "st,stm32mp25-omi";
+				reg = <0 0 0x400>;
+				clocks = <&rcc CK_KER_OSPI1>;
+				resets = <&rcc OSPI1_R>, <&rcc OSPI1DLL_R>;
+				status = "disabled";
+			};
+
+			ospi2: spi@40440000 {
+				compatible = "st,stm32mp25-omi";
+				reg = <1 0 0x400>;
+				clocks = <&rcc CK_KER_OSPI2>;
+				resets = <&rcc OSPI2_R>, <&rcc OSPI2DLL_R>;
+				status = "disabled";
+			};
+		};
+
 		rifsc: rifsc@42080000 {
-			compatible = "st,stm32mp25-rifsc";
+			compatible = "st,stm32mp25-rifsc", "syscon";
 			reg = <0x42080000 0x1000>;
 			#address-cells = <1>;
 			#size-cells = <1>;
@@ -218,6 +248,30 @@
 				status = "disabled";
 			};
 
+			hash: hash@42010000 {
+				compatible = "st,stm32mp13-hash";
+				reg = <0x42010000 0x400>;
+				clocks = <&rcc CK_BUS_HASH>;
+				resets = <&rcc HASH_R>;
+				status = "disabled";
+			};
+
+			rng: rng@42020000 {
+				compatible = "st,stm32mp13-rng";
+				reg = <0x42020000 0x400>;
+				clocks = <&rcc CK_BUS_RNG>;
+				resets = <&rcc RNG_R>;
+				status = "disabled";
+			};
+
+			iwdg1: watchdog@44010000 {
+				compatible = "st,stm32mp2-iwdg";
+				reg = <0x44010000 0x400>;
+				clocks = <&rcc CK_BUS_IWDG1>, <&rcc LSI_CK>;
+				clock-names = "pclk", "lsi";
+				status = "disabled";
+			};
+
 			i2c8: i2c@46040000 {
 				compatible = "st,stm32mp25-i2c";
 				reg = <0x46040000 0x400>;
@@ -225,54 +279,34 @@
 				resets = <&rcc I2C8_R>;
 				status = "disabled";
 			};
-		};
-
-		ommanager: ommanager@40500000 {
-			#address-cells = <2>;
-			#size-cells = <1>;
-			compatible = "st,stm32mp25-omm";
-			reg = <0x40500000 0x400>, <0x60000000 0x10000000>;
-			reg-names = "omm", "omm_mm";
-			clocks = <&rcc CK_BUS_OSPIIOM>;
-			resets = <&rcc OSPIIOM_R>;
-			status = "disabled";
 
-			ranges = <0 0 0x40430000 0x400>,
-				 <1 0 0x40440000 0x400>;
-
-			ospi1: spi@40430000 {
-				compatible = "st,stm32mp25-omi";
-				reg = <0 0 0x400>;
-				clocks = <&rcc CK_KER_OSPI1>;
-				resets = <&rcc OSPI1_R>, <&rcc OSPI1DLL_R>;
+			sdmmc1: mmc@48220000 {
+				compatible = "st,stm32mp25-sdmmc2", "arm,pl18x", "arm,primecell";
+				arm,primecell-periphid = <0x00353180>;
+				reg = <0x48220000 0x400>, <0x44230400 0x8>;
+				clocks = <&rcc CK_KER_SDMMC1>;
+				clock-names = "apb_pclk";
+				resets = <&rcc SDMMC1_R>;
+				cap-sd-highspeed;
+				cap-mmc-highspeed;
+				max-frequency = <120000000>;
 				status = "disabled";
 			};
 
-			ospi2: spi@40440000 {
-				compatible = "st,stm32mp25-omi";
-				reg = <1 0 0x400>;
-				clocks = <&rcc CK_KER_OSPI2>;
-				resets = <&rcc OSPI2_R>, <&rcc OSPI2DLL_R>;
+			sdmmc2: mmc@48230000 {
+				compatible = "st,stm32mp25-sdmmc2", "arm,pl18x", "arm,primecell";
+				arm,primecell-periphid = <0x00353180>;
+				reg = <0x48230000 0x400>, <0x44230800 0x8>;
+				clocks = <&rcc CK_KER_SDMMC2>;
+				clock-names = "apb_pclk";
+				resets = <&rcc SDMMC2_R>;
+				cap-sd-highspeed;
+				cap-mmc-highspeed;
+				max-frequency = <120000000>;
 				status = "disabled";
 			};
 		};
 
-		hash: hash@42010000 {
-			compatible = "st,stm32mp13-hash";
-			reg = <0x42010000 0x400>;
-			clocks = <&rcc CK_BUS_HASH>;
-			resets = <&rcc HASH_R>;
-			status = "disabled";
-		};
-
-		rng: rng@42020000 {
-			compatible = "st,stm32mp13-rng";
-			reg = <0x42020000 0x400>;
-			clocks = <&rcc CK_BUS_RNG>;
-			resets = <&rcc RNG_R>;
-			status = "disabled";
-		};
-
 		risaf2: risaf@420b0000 {
 			compatible = "st,stm32-risaf";
 			reg = <0x420b0000 0x1000>;
@@ -322,14 +356,6 @@
 			};
 		};
 
-		iwdg1: watchdog@44010000 {
-			compatible = "st,stm32mp2-iwdg";
-			reg = <0x44010000 0x400>;
-			clocks = <&rcc CK_BUS_IWDG1>, <&rcc LSI_CK>;
-			clock-names = "pclk", "lsi";
-			status = "disabled";
-		};
-
 		rcc: rcc@44200000 {
 			compatible = "st,stm32mp25-rcc";
 			reg = <0x44200000 0x10000>;
@@ -403,32 +429,6 @@
 			};
 		};
 
-		sdmmc1: mmc@48220000 {
-			compatible = "st,stm32mp25-sdmmc2", "arm,pl18x", "arm,primecell";
-			arm,primecell-periphid = <0x00353180>;
-			reg = <0x48220000 0x400>, <0x44230400 0x8>;
-			clocks = <&rcc CK_KER_SDMMC1>;
-			clock-names = "apb_pclk";
-			resets = <&rcc SDMMC1_R>;
-			cap-sd-highspeed;
-			cap-mmc-highspeed;
-			max-frequency = <120000000>;
-			status = "disabled";
-		};
-
-		sdmmc2: mmc@48230000 {
-			compatible = "st,stm32mp25-sdmmc2", "arm,pl18x", "arm,primecell";
-			arm,primecell-periphid = <0x00353180>;
-			reg = <0x48230000 0x400>, <0x44230800 0x8>;
-			clocks = <&rcc CK_KER_SDMMC2>;
-			clock-names = "apb_pclk";
-			resets = <&rcc SDMMC2_R>;
-			cap-sd-highspeed;
-			cap-mmc-highspeed;
-			max-frequency = <120000000>;
-			status = "disabled";
-		};
-
 		/*
 		 * Break node order to solve dependency probe issue between
 		 * pinctrl and exti.
-- 
2.53.0

